qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/RollingUpgradeLuceneIndexCompatibilityTestCase.java

```diff
@@ -194,6 +249,15 @@ public void testRestoreIndex() throws Exception {
                 closeIndex(restoredIndex);
                 ensureGreen(restoredIndex);
 
+                logger.debug("--> write API block can be removed on a closed index: INDEX_CLOSED_BLOCK already blocks writes");
```

Comment:
Can we also check that if an index has both read-only and write block then we can remove one of them for an N-2 index?

Reply:
I reworked the testIndexUpgrade test to randomly add one or two of the blocks, and then remove one of them.

See 59e7acddb8addaeda32172a3ae90ece223f6cde4